mlflow/utils/rest_utils.py

```diff
@@ -36,7 +37,9 @@ def http_request(host_creds, endpoint, retries=3, retry_interval=3, **kwargs):
     elif host_creds.token:
         auth_str = "Bearer %s" % host_creds.token
 
-    headers = {}
+    headers = {
+        'User-Agent': 'mlflow-python-client/%s' % __version__,
```

Comment:
Had been wondering whether it'd make sense to separate client type & version into separate headers (to e.g. simplify grouping requests by client type across different versions), but putting client & version info into a single User-Agent header seems standard ([per MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent)) & it doesn't seem like there are other clear alternatives ([link](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#Request_context))